src/typed_data.rs

```diff
@@ -256,9 +256,18 @@ where
         if self.free_immediately {
             flags |= RUBY_TYPED_FREE_IMMEDIATELY as VALUE;
         }
-        if self.wb_protected {
```

Comment:
There's no need to wrap this in an option, the builder only lets you enable flags, so 'false' and 'unset' are the same. You can just change this condition to `self.wb_protected || !self.mark`.

Reply:
I was originally thinking we may want to support opting this default behavior, in which case the option makes sense. But happy to adjust if you don’t envision that as a use case.

Reply:
Yeah, I don't see needing to let people opt out, especially as the Ruby change you linked would just override that opt out anyway.

Reply:
True. I’ll update.